sqldb/migrations.go

```diff
@@ -381,6 +385,29 @@ func ApplyMigrations(ctx context.Context, db *BaseDB,
 			"(dirty=%v) as base version", currentVersion, dirty)
 	}
 
+	// Due to an a migration issue in v0.19.0-rc1 we may be at version 2 and
+	// have a dirty schema due to failing migration 3. If this is indeed the
+	// case, we need to reset the dirty flag to be able to apply the fixed
+	// migration.
+	if version == 2 {
+		schemaVersion, dirty, err := migrator.GetSchemaVersion()
+		if err != nil {
+			return err
+		}
+
```

Comment:
Should we return an error here if we encounter the version 2 and no dirty bit set ?

```
if !dirty {
  return errors.New("did expect the db to be in a dirty state")
} 
```

can we phase out this code in the future, we should add a comment to do so ?

Reply:
No, because if the schema isn't dirty, the migration should proceed as normal. Added a comment for phase out.